elastic/apm-server
# feat: auto set GOMEMLIMIT when memory is limited

## Motivation/summary

By default memlimit uses all the memory available on the host but if cgroups are used and the process is being limited we want to use the correct max memory.
Use automemlimit lib to set memory limit
Run check every 30s
Similar approach as automaxprocs 

Test has to be performed manually (see below) because we can't really restrict the process memory :(

## Checklist




For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

- `make apm-server`
- `docker run -it -v ./:/apm-server --memory=4096m golang`
- cd /apm-server && ./apm-server -e
- observe the the log line changing the memlimit


## Related issues

Closes https://github.com/elastic/apm-server/issues/14475
